sensor.py

```diff
@@ -335,3 +348,59 @@ def _handle_coordinator_update(self):
 
         self._attr_is_on = json["onOffStatus"]["parNewVal"] == "1"
         self.async_write_ha_state()
+
+
+class HonWashingMachineMode(SensorEntity, HonWashingMachineEntity):
+    def __init__(self, hass, coordinator, entry, appliance) -> None:
+        super().__init__(hass, entry, coordinator, appliance)
+
+        self._coordinator = coordinator
+        self._attr_unique_id = f"{self._mac}_machine_mode"
+        self._attr_name = f"{self._name} Machine Mode"
+        self._attr_icon = "mdi:chef-hat"
+
+    @callback
+    def _handle_coordinator_update(self):
+
+        # Get state from the cloud
+        json = self._coordinator.data
+
+        # No data returned by the Get State method (unauthorized...)
+        if json is False:
+            return
+
+        mode = json["machMode"]
```

Comment:
I tested the PR using Washing machine: `HWE 49AMBS/1-S`
json data contains `"machMode": {"parNewVal": "2", "lastUpdate": "2022-08-01T09:49:16Z"}`

Worked after updating sensor.py  to 
```python
mode = json["machMode"]["parNewVal"]
```

Reply:
Thanks, I will review this next week.